cmd/snap/cmd_buy_test.go

```diff
@@ -73,6 +73,9 @@ func (s *buyTestMockSnapServer) checkCounts() {
 }
 
 func (s *SnapSuite) TestBuyHelp(c *check.C) {
+	s.Login(c)
```

Comment:
This looks a lot like you want to define your own BuySnapSuite on top of SnapSuite and run s.Login(c) in SetUpTest() there.